evm/src/main/java/org/hyperledger/besu/evm/frame/Memory.java

```diff
@@ -202,11 +204,40 @@ public Bytes getBytes(final long location, final long numBytes) {
     }
 
     final int start = asByteIndex(location);
-
     ensureCapacityForBytes(start, length);
     return Bytes.wrap(Arrays.copyOfRange(memBytes, start, start + length));
   }
 
+  /**
+   * Returns a copy of bytes by peeking into memory without expanding the active words.
+   *
+   * @param location The location in memory to start with.
+   * @param numBytes The number of bytes to get.
+   * @return A fresh copy of the bytes from memory starting at {@code location} and extending {@code
+   *     numBytes}.
+   */
+  public Bytes getBytesWithoutGrowth(final long location, final long numBytes) {
+    // Note: if length == 0, we don't require any memory expansion, whatever location is. So
+    // we must call asByteIndex(location) after this check so as it doesn't throw if the location
+    // is too big but the length is 0 (which is somewhat nonsensical, but is exercise by some
+    // tests).
+    final int length = asByteLength(numBytes);
+    if (length == 0) {
+      return Bytes.EMPTY;
+    }
+
+    final int start = asByteIndex(location);
+
+    // Arrays.copyOfRange would throw if start > memBytes.length, so just return the expected
+    // number of zeros without expanding the memory.
+    // Otherwise, just follow the happy path.
+    if (start > memBytes.length) {
+      return Bytes.wrap(new byte[(int) numBytes]);
+    } else {
```

Comment:
This doesn't handle a "straddle" case, where part of the bytes are in unexpanded memory and the rest needs to be zero expanded.  I would expect that a call with `n` as `numBytes` should allow a `get(n-1)` to work as well as `getSize()` on the return result retuning `n`

See https://github.com/hyperledger/besu/blob/98a06e882c68c5a24c55f7e67f14662b320cdb5f/evm/src/main/java/org/hyperledger/besu/evm/precompile/BigIntegerModularExponentiationPrecompiledContract.java#L211-L222

Reply:
According to [`Arrays.copyOfRange` doc](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#copyOfRange-byte:A-int-int-) handles this case:

> may be greater than original.length, in which case (byte)0 is placed in all elements of the copy whose index is greater than or equal to original.length - from.